pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java

```diff
@@ -461,11 +465,28 @@ public void getVersionBySchema(
     }
 
     private static GetSchemaResponse convertSchemaAndMetadataToGetSchemaResponse(SchemaAndMetadata schemaAndMetadata) {
+        String schemaData;
+        if (schemaAndMetadata.schema.getType() == SchemaType.KEY_VALUE) {
+            ByteBuf byteBuf = ByteBufAllocator.DEFAULT.heapBuffer().writeBytes(schemaAndMetadata.schema.getData());
```

Comment:
I think KeyValueSchemaInfo or KeyValueSchema provides the until functions to split the key schema and value schema. We should try to use those util functions rather than reimplementing the logic here again.